config/schemas/config.yaml

```diff
@@ -0,0 +1,2854 @@
+    allOf:
+      - $ref: '#/$defs/commonResource'
+      - type: object
+        properties:
+          cainjector:
+            $ref: '#/$defs/commonResource'
+          extraArgs:
+            $ref: '#/$defs/extraArgs'
+          webhook:
+            $ref: '#/$defs/commonResource'
+  clusterAdmin:
+    properties:
+      groups:
+        type: array
+        items:
+          type: string
+      users:
+        items:
+          type: string
+        type: array
+    type: object
+  clusterApi:
+    properties:
+      clusters:
+        items:
+          $ref: '#/$defs/clusterid'
+        type: array
+      enabled:
+        default: false
+        type: boolean
+      monitoring:
+        properties:
+          enabled:
+            default: false
+            type: boolean
+        type: object
+    type: object
+  dex:
+    properties:
+      additionalKubeloginRedirects:
+        type: array
+      affinity:
+        $ref: '#/$defs/affinity'
+      enableStaticLogin:
+        default: true
+        type: boolean
+      expiry:
+        properties:
+          deviceRequests:
+            default: 5m
+            type: string
+          idToken:
+            default: 24h
+            type: string
+          refreshTokens:
+            properties:
+              absoluteLifetime:
+                default: 3960h
+                type: string
+              reuseInterval:
+                default: 3s
+                type: string
+              validIfNotUsedFor:
+                default: 2160h
+                type: string
+            type: object
+          signingKeys:
+            default: 6h
+            type: string
+        type: object
+      google:
+        properties:
+          groupSupport:
+            default: false
+            type: boolean
+        type: object
+      nodeSelector:
+        $ref: '#/$defs/nodeSelector'
+      replicaCount:
+        default: 2
+        type: number
+      resources:
+        $ref: '#/$defs/resources'
+      serviceMonitor:
+        properties:
+          enabled:
+            default: true
+            type: boolean
+        type: object
+      subdomain:
+        default: dex
+        type: string
+      tolerations:
+        $ref: '#/$defs/tolerations'
+      topologySpreadConstraints:
+        items:
+          properties:
+            labelSelector:
+              properties:
+                matchLabels:
+                  properties:
+                    app.kubernetes.io/name:
+                      default: dex
+                      type: string
+                  type: object
+              type: object
+            maxSkew:
+              default: 1
+              type: number
+            topologyKey:
+              default: kubernetes.io/hostname
+              type: string
+            whenUnsatisfiable:
+              default: DoNotSchedule
+              type: string
+          type: object
+        type: array
```

Comment:
Is it possible to extract these as a common thing?

Do one loose the ability to set default values when doing that?

And would it be good to drive default values through the schema, given that we have different profiles with different default values?

Reply:
There was a common `topologySpreadConstraints` already, pointed all of them to the same.